os/mm/kmm_heap/kmm_calloc.c

```diff
@@ -83,7 +83,13 @@ static void *kheap_calloc(size_t n, size_t elem_size, size_t retaddr)
 			return ret;
 		}
 	}
+#ifdef CONFIG_MM_ASSERT_ON_FAIL
+	lldbg("Allocation failed from kernel heap.\n");
+#else
+	mdbg("Allocation failed from kernel heap.\n");
+#endif
```

Comment:
Since there is a repetition of the above logs, how about moving these to ` mm_manage_alloc_fail` 
Something like this
```void mm_manage_alloc_fail(mm_heap_s *heap, heap_start_index, number_of_heaps, bool is_kernel_heap , size);```
so that we can print these logs accordingly, if the number of heaps is 1, we can print the lldbg("Allocation failed from %d-th user/kernel heap.\n", heap_index);
if greater than 1 we can print lldbg("Allocation failed from user/kernel heap.\n");

Reply:
But in this case, we cannot distinguish that heap is from kernel or user.
So I added in here.

Reply:
we can add an argument to mm_manage_alloc_fail to determine if it is user heap or kernel heap as I mentioned in the comment above. Please see if this helps.